koku/api/status/model.py

```diff
@@ -0,0 +1,104 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+"""Models to capture server status."""
+
+import logging
+import os
+import platform
+import subprocess
+import sys
+import uuid
+
+from django.db import models
+
+from api import API_VERSION
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+class Status(models.Model):
+    """A server's status."""
+
+    server_id = models.UUIDField(default=uuid.uuid4,
+                                 editable=False)
+
+    def get_commit(self):  # pylint: disable=R0201
+        """Collect the build number for the server.
+
+        :returns: A build number
+        """
+        commit_info = os.environ.get('KOKU_COMMIT', None)
+        if commit_info is None:
+            try:
+                commit_info = subprocess.check_output(['git',
+                                                       'describe',
+                                                       '--always'])
+                commit_info = commit_info.decode('utf-8').strip()
+            except subprocess.CalledProcessError:
```

Comment:
I still feel funny about this being in a try/except where the except is just a pass. `subprocess.run()` should usable with the check flag set to False so that an exception is not raised.

Reply:
updated.